http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -582,51 +582,55 @@ class LeoPubsubMessageSubscriber[F[_]](
   private[monitor] def createDisk(msg: CreateDiskMessage, formattedBy: Option[FormattedBy], sync: Boolean)(implicit
     ev: Ask[F, AppContext]
   ): F[Unit] = {
-    val create = for {
-      ctx <- ev.ask
-      operationFutureOpt <- googleDiskService
-        .createDisk(
-          msg.googleProject,
-          msg.zone,
-          Disk
-            .newBuilder()
-            .setName(msg.name.value)
-            .setSizeGb(msg.size.gb)
-            .setZone(msg.zone.value)
-            .setType(msg.diskType.googleString(msg.googleProject, msg.zone))
-            .setPhysicalBlockSizeBytes(msg.blockSize.bytes)
-            .putAllLabels(Map("leonardo" -> "true").asJava)
-            .build()
-        )
-      _ <- operationFutureOpt match {
-        case None => F.unit
-        case Some(v) =>
-          val task = for {
-            _ <- F.blocking(v.get())
-            _ <- formattedBy match {
-              case Some(value) =>
-                persistentDiskQuery
-                  .updateStatusAndIsFormatted(msg.diskId, DiskStatus.Ready, value, ctx.now)
-                  .transaction[F]
-                  .void
-              case None =>
-                persistentDiskQuery.updateStatus(msg.diskId, DiskStatus.Ready, ctx.now).transaction[F].void
-            }
-          } yield ()
+    val create = {
+      val diskBuilder = Disk
+        .newBuilder()
+        .setName(msg.name.value)
+        .setSizeGb(msg.size.gb)
+        .setZone(msg.zone.value)
+        .setType(msg.diskType.googleString(msg.googleProject, msg.zone))
+        .setPhysicalBlockSizeBytes(msg.blockSize.bytes)
+        .putAllLabels(Map("leonardo" -> "true").asJava)
+      msg.sourceDisk.foreach(sourceDisk => diskBuilder.setSourceDisk(sourceDisk.asString))
```

Comment:
functionally it's the same here..but can we use `traverse` instead of `foreach`

Also, what does the string format of `DiskLink` look like? it'd be nice to add at `DiskLink`'s definition site

Reply:
If I use `traverse` instead I need to move it down into the for comp and do
```
        _ <- msg.sourceDisk.traverse(sourceDisk => F.pure(diskBuilder.setSourceDisk(sourceDisk.asString))) 
```
`foreach` is more straight forward to me but whatever you want.

I don't really know what the format of `DiskLink` looks like. It comes from google and we send it to google. 🙈